core/tests/test_block.h

```diff
@@ -0,0 +1,297 @@
+// -------------------------------------------------------------
+//  Cubzh Core
+//  block.h
+//  Created by Nino PLANE on October 19, 2022.
+// -------------------------------------------------------------
+
+#pragma once
+
+#include "block.h"
+
+void test_block_new(void) {
+    Block* a = block_new();
+
+    TEST_CHECK(a->colorIndex == 0);
+    block_free(a);
+}
+
+void test_block_new_air(void) {
+    Block* a = block_new_air();
+
+    TEST_CHECK(a->colorIndex == 255);
+    block_free(a);
+}
+
+void test_block_new_with_color(void) {
+    SHAPE_COLOR_INDEX_INT_T color = 152;
+
+    Block *a = block_new_with_color(color);
+    TEST_CHECK(a->colorIndex == 152);
+    block_free(a);
+}
+
+void test_block_new_copy(void) {
+    SHAPE_COLOR_INDEX_INT_T color = 152;
+    Block *a = block_new_with_color(color);
+    Block *b = block_new();
+
+    Block *c = block_new_copy(a);
+    TEST_CHECK(c->colorIndex == 152);
+    c = block_new_copy(b);
+    TEST_CHECK(c->colorIndex == 0);
+    block_free(a);
+    block_free(b);
+    block_free(c);
+}
+
+void test_block_set_color_index(void) {
+    Block* a = block_new();
+    SHAPE_COLOR_INDEX_INT_T color = 152;
+
+    block_set_color_index(a, color);
+    TEST_CHECK(a->colorIndex == 152);
+    block_free(a);
+}
+
+void test_block_get_color_index(void) {
+    Block* a = block_new();
+    Block* b = block_new_air();
+    SHAPE_COLOR_INDEX_INT_T color = 152;
+    SHAPE_COLOR_INDEX_INT_T colorCheck;
```

Comment:
This variable is not need until later in the test.
You can move it down at least 1 line.